src/neural/cuda/network_cudnn.cc

```diff
@@ -272,12 +276,36 @@ class CudnnNetwork : public Network {
     // support it).
     ReportCUBLASErrors(cublasSetMathMode(cublas_, CUBLAS_TENSOR_OP_MATH));
 
-    // Check if we want to enable our custom winograd ?
+
     constexpr bool fp16 = std::is_same<half, DataType>::value;
+    const int kNumInputPlanes = kInputPlanes;
+    const int kNumFilters = weights.input.biases.size();
+    numBlocks_ = weights.residual.size();
+
+    // Use our custom winograd for resudual tower convolutions for most cases:
+    //
+    //  1. Should be always faster than cudnn's winograd that we use for fp32,
+    //  and for fp16 on GPUs without tensor cores
+    // 
+    //  2. Should also be faster than cudnn's implicit GEMM on GPUs with tensor 
+    //     cores too, but only for networks with 256 or higher no. of filters.
+    // 
+    //  3. Currently a bug in cublas makes it slower on RTX GPUs with fp16 so
+    //  it's disabled. TODO: Enable it once the bug has been fixed and it's
+    //  tested to be faster.
+
+    if (fp16) {
+      if (!hasTensorCores)
+        use_custom_winograd_ = true;
+      else if (kNumFilters >= 256 && deviceProp.minor != 5) // TODO: remove the second condition.
```

Comment:
Is it safe to check minor without a major check to go with it?

Reply:
Also can we add a check here that cublas version <= 10202 (the currently released one)? I'm assuming that the next release will have the fix and that it won't be ported to cuda 10.1 etc..

Reply:
Thanks. modified the check.